src/styles/blocks/calender.scss

```diff
@@ -0,0 +1,44 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $day-gap;
+  max-width: (7 * $day-side) + (14 * $day-gap) + (2 * $calendar-padding);
+  padding: $calendar-padding;
+
+  @each $day, $index in $days {
+    &--start-day--#{$day} :first-child {
+      margin-left: ($day-side * $index) + ($day-gap * $index);
+    }
+  }
+
+  @for $lastDate from 28 through 31 {
+    &--month-length--#{$lastDate} {
+      .calendar__day:nth-child(n + #{$lastDate + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__day {
+    @for $day from 1 through 31 {
+      &--#{$day}::before {
+        content: "#{$day}";
+      }
+    }
+
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    border: 1px solid $black;
+    height: $day-side;
+    width: $day-side;
+    background-color: $grey;
+
+    &:hover {
+      transform: translateY(-20px);
+      background-color: $pink;
+      transition: background-color 0.5s, transform 0.5s;
```

Comment:
transition shall be used on element, instead of :hover - see [checklist](https://github.com/mate-academy/layout_calendar/blob/master/checklist.md#:~:text=transition%20style%20under%20general%20selector%2C%20not%20the%20one%20with%20%3Ahover)